backend/app/schemas/admin_users.py

```
@@ -0,0 +1,66 @@
+class AdminCreateUserRequest(BaseModel):
+    username: str = Field(min_length=3, max_length=50)
+    email: str = Field(min_length=5, max_length=255)
+    password: str | None = Field(default=None, min_length=8, max_length=256)
+    generate_password: bool = False
+    is_admin: bool = False
+    is_active: bool = True
+
+    @model_validator(mode="after")
+    def validate_password_source(self) -> "AdminCreateUserRequest":
+        if not self.generate_password and not self.password:
+            raise ValueError("Either password must be provided or generate_password must be true")
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)